x/mm2/types/params.go

```diff
@@ -0,0 +1,59 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	// DefaultVersion is the default value for the Version Param.
+	DefaultVersion = 0
+)
+
+// DefaultMarketAuthority is the default value for the market authority Param.
+var DefaultMarketAuthority = authtypes.NewModuleAddress(govtypes.ModuleName).String()
+
+// DefaultParams returns default marketmap parameters.
+func DefaultParams() Params {
+	return Params{
+		MarketAuthorities: []string{DefaultMarketAuthority},
+		Version:           DefaultVersion,
+	}
+}
+
+// NewParams returns a new Params instance.
+func NewParams(authorities []string, version uint64) (Params, error) {
+	if authorities != nil {
+		return Params{}, fmt.Errorf("cannot create Params with nil authority")
+	}
+
+	return Params{
+		MarketAuthorities: authorities,
+		Version:           version,
+	}, nil
+}
+
+// ValidateBasic performs stateless validation of the Params.
+func (p *Params) ValidateBasic() error {
+	if p.MarketAuthorities == nil {
+		return fmt.Errorf("cannot create Params with nil authority")
+	}
+
+	seenAuthorities := make(map[string]struct{}, len(p.MarketAuthorities))
```

Comment:
we perform this check so frequently (uniqueness of items in an array), perhaps it's useful for us to make a generic helper in `/lib` and start using that

Reply:
https://linear.app/skip/issue/BLO-1059/create-lib-function-for-checking-uniqueness-of-items-in-a-list